plugins/woocommerce-blocks/assets/js/base/components/select/index.tsx

```diff
@@ -22,61 +27,151 @@ type SelectProps = Omit<
+		} else {
+			setValidationErrors( {
+				[ errorIdString ]: {
+					message: errorMessage,
+					hidden: true,
+				},
+			} );
+		}
+		return () => {
+			clearValidationError( errorIdString );
+		};
+	}, [
+		clearValidationError,
+		value,
+		errorIdString,
+		errorMessage,
+		required,
+		setValidationErrors,
+	] );
+
+	const validationError = useSelect( ( select ) => {
+		const store = select( VALIDATION_STORE_KEY );
+		return (
+			store.getValidationError( errorIdString || '' ) || {
+				hidden: true,
+			}
+		);
+	} );
+
 	const generatedId = useId();
 
 	const inputId =
 		restOfProps.id || `wc-blocks-components-select-${ generatedId }`;
 
 	return (
-		<div className={ `wc-blocks-components-select ${ className || '' }` }>
-			<div className="wc-blocks-components-select__container">
-				<label
-					htmlFor={ inputId }
-					className="wc-blocks-components-select__label"
-				>
-					{ label }
-				</label>
-				<select
-					className="wc-blocks-components-select__select"
-					id={ inputId }
-					size={ size !== undefined ? size : 1 }
-					onChange={ selectOnChange }
-					value={ value }
-					{ ...restOfProps }
-				>
-					{ options.map( ( option ) => (
-						<option
-							key={ option.value }
-							value={ option.value }
-							data-alternate-values={ `[${ option.label }]` }
-							disabled={
-								option.disabled !== undefined
-									? option.disabled
-									: false
-							}
-						>
-							{ option.label }
-						</option>
-					) ) }
-				</select>
-				<Icon
-					className="wc-blocks-components-select__expand"
-					icon={ chevronDown }
-				/>
+		<div
+			className={ clsx( className, {
+				'has-error': ! validationError.hidden,
+			} ) }
+		>
+			<div className="wc-blocks-components-select">
+				<div className="wc-blocks-components-select__container">
+					<label
+						htmlFor={ inputId }
+						className="wc-blocks-components-select__label"
+					>
+						{ label }
+					</label>
+					<select
+						className="wc-blocks-components-select__select"
+						id={ inputId }
+						size={ size !== undefined ? size : 1 }
+						onChange={ selectOnChange }
+						value={ value }
+						aria-invalid={
+							error?.message && ! error?.hidden ? true : false
+						}
+						aria-errormessage={ validationErrorId }
+						{ ...restOfProps }
+					>
+						{ optionsWithEmpty.map( ( option ) => (
+							<option
+								key={ option.value }
+								value={ option.value }
+								data-alternate-values={ `[${ option.label }]` }
+								disabled={
+									option.disabled !== undefined
+										? option.disabled
+										: false
+								}
+							>
+								{ option.label }
+							</option>
+						) ) }
+					</select>
+					<Icon
+						className="wc-blocks-components-select__expand"
+						icon={ chevronDown }
+					/>
+				</div>
 			</div>
+			{ validationError && validationError.hidden !== true && (
+				<ValidationInputError
+					errorMessage={ validationError.message }
+				/>
+			) }
```

Comment:
I could have done

```jsx
	<ValidationInputError propertyName={ errorId } />
```

But I didn't to match the original select.

Reply:
@senadir why is it so important to match the original? We know there are a few bugs and things done incorrectly lying around.

Reply:
When I say original Select I meant the select component you used in country and state.

For the combobox, it uses `ValidationInputError` with `propertyName`, the new country and state select, it uses the above. I didn't want to modify your code because I wanted to keep this a light refactor.

Reply:
Updated.

Reply:
@samueljseay do you recall why you did:

```jsx
{ validationError && validationError.hidden !== true && (
	<ValidationInputError
		errorMessage={ validationError.message }
	/>
) }
```

instead of:

```jsx
<ValidationInputError propertyName={ errorId } />
```

Reply:
@senadir oh whoops that was a mistake on my part when I was using `ValidationInputError` I passed in errorMessage (I was having trouble getting the select to find the validation message for some reason) and then I didn't refactor this after I got that working. I realize now that is handled internally. The API of that component is a little confusing.

Reply:
It is confusing indeed, you're the second one getting confused by it this week.